src/modules/0.0.22/aws_rds/index.ts

```diff
@@ -260,26 +268,30 @@ class RdsMapper extends MapperBase<RDS> {
       return out;
     },
     read: async (ctx: Context, id?: string) => {
-      const client = (await ctx.getAwsClient()) as AWS;
+      const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
       if (id) {
-        const rawRds = await this.getDBInstance(client.rdsClient, id);
+        const { dbInstanceIdentifier, region } = this.idFields(id);
+        const client = (await ctx.getAwsClient(region)) as AWS;
+        const rawRds = await this.getDBInstance(client.rdsClient, dbInstanceIdentifier);
         if (!rawRds) return;
-        return await this.rdsMapper(rawRds, ctx);
+        return await this.rdsMapper(rawRds, ctx, region);
       } else {
-        const rdses = await this.getDBInstances(client.rdsClient);
         const out = [];
-        for (const rds of rdses) {
-          out.push(await this.rdsMapper(rds, ctx));
+        for (const region of enabledRegions) {
+          const client = (await ctx.getAwsClient(region)) as AWS;
+          const rdses = await this.getDBInstances(client.rdsClient);
+          for (const rds of rdses) {
+            out.push(await this.rdsMapper(rds, ctx, region));
+          }
```

Comment:
let's do this in parallel with `await.Promise.all()` like here https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_api_gateway/index.ts#L79L91

Reply:
That inner `.map` should be a `.forEach`, but got it.